basic_sender.go

```diff
@@ -186,8 +187,34 @@ func NewLogger(ctx context.Context, name string, l send.LevelInfo, opts *LoggerO
 	return b, nil
 }
 
-// MakeLogger returns a grip Sender backed by Cedar Buildlogger.
-func MakeLogger(ctx context.Context, name string, opts *LoggerOptions) (send.Sender, error) {
+// NewLogger returns a grip Sender backed by cedar Buildlogger with level
+// information set and uses passed in context.
+func NewLoggerWithContext(ctx context.Context, name string, l send.LevelInfo, opts *LoggerOptions) (send.Sender, error) {
+	b, err := MakeLoggerWithContext(ctx, name, opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem making new logger")
+	}
+
+	if err := b.SetLevel(l); err != nil {
+		return nil, errors.Wrap(err, "problem setting grip level")
+	}
+
+	return b, nil
+}
+
+// MakeLogger returns a grip Sender backed by cedar Buildlogger.
+func MakeLogger(name string, opts *LoggerOptions) (send.Sender, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	return makeLogger(ctx, cancel, name, opts)
+}
+
+// MakeLoggerWithContext returns a grip Sender backed by cedar Buildlogger and
+// uses the passed in context.
+func MakeLoggerWithContext(ctx context.Context, name string, opts *LoggerOptions) (send.Sender, error) {
+	return makeLogger(ctx, nil, name, opts)
+}
+
+func makeLogger(ctx context.Context, cancel context.CancelFunc, name string, opts *LoggerOptions) (send.Sender, error) {
```

Comment:
I'd not pass the cancel func, and just call WithCancel at the top of this function.